db/migrate/20181203003146_create_users.rb

```diff
@@ -0,0 +1,18 @@
+class CreateUsers < ActiveRecord::Migration[5.2]
+  def change
+    create_table :users do |t|
```

Comment:
By default, Rails is setup to implement IDs as `integer`s in Postgres.

**Should we consider UUIDs instead?**
https://andycroll.com/ruby/choose-uuids-for-model-ids-in-rails/


> Using UUIDs as the id in your Rails models instead of incrementing integers helps you avoid collisions. The UUIDs are globally unique meaning you can know that different models cannot possibly have the same id and you can even assign them client-side or in other systems.

> With an incrementing integer id the size of your data can be inferred from the outside i.e. id 5 is the fifth record created. With UUIDs no-one can guess the size of your database tables, which might be information you are keen to keep secret. You can get round this by generating ‘public ids’ or ‘slugs’ for exposed URLs… but then, why not use a built-in tool?

> From a security perspective, using UUIDs prevents the situation where a malicious attacker could attempt to gain access to data by guessing a model id in your URLs. UUIDs are extremely hard to guess.

> This is a case where you are making a choice toward a little more complexity, but for good reasons.

There's a section in there which also mentions the "why not"s, but I think the "security through obscurity" piece here is pretty nice, and it's super easy to setup while we're here:

https://blog.bigbinary.com/2016/04/04/rails-5-provides-application-config-to-use-uuid-as-primary-key
https://lab.io/articles/2017/04/13/uuids-rails-5-1/

There are a few other articles out there - differences might exist based on PSG versions.

----

This kind of thing ends up being a bit more work, but it does tend to make things like integrating with microservices a bit easier. Non-integer UUIDs tend to be easier to work with as it's damn near impossible for data collisions to occur.

I'm OK with either approach - if we want to go to UUIDs, now would be the best time.

Reply:
Done!